sdk/storage/storage-file-datalake/src/clients.ts

```diff
@@ -329,13 +329,33 @@ export class DataLakePathClient extends StorageClient {
     const { span, updatedOptions } = createSpan("DataLakePathClient-create", options);
     try {
       ensureCpkIfSpecified(options.customerProvidedKey, this.isHttps);
+      let expiryOptions: FileExpiryMode | undefined;
+      let expiresOn: string | undefined;
+      if (options.expiresOn) {
+        if (Number.isFinite(options.expiresOn)) {
+          expiryOptions = "RelativeToNow";
+          expiresOn = Math.round(options.expiresOn! as number).toString();
+        } else {
+          const fileExpiryTime = options.expiresOn as Date;
```

Comment:
This can not work as check for Date type.

Reply:
Updated